File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> valori;
        for(int i=0; i < nums.size(); ++i){
            if(valori.count(target - nums[i]))
                return {valori[target - nums[i]], i};
            valori[nums[i]] = i;
            
        }
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        suma, minim, left, size = 0, 0, 0, 0
        for i in range(len(nums)):
            suma += nums[i]
            size += 1
            if suma >= target:
                if minim == 0:
                    minim = size
                while True:
                    if suma - nums[left] < target:
                        if minim > size:
                            minim = size
                        break
                    suma -= nums[left]
                    left += 1
                    size -= 1

        return minim